sympy/solvers/ode/riccati.py

```diff
@@ -283,6 +303,42 @@ def construct_d(a, x, val_inf):
     return d
 
 
+def rational_laurent_series(num, den, x, x0, mul, n):
+    m = Symbol('m')
+    one = Poly(1, x)
+    reverse = False
+    if x0 == oo:
+        num, den = inverse_transform_poly(num, den, x)
+        reverse = True
+        x0 = 0
+    if x0:
+        num = num.transform(Poly(x + x0, x), one)
+        den = den.transform(Poly(x + x0, x), one)
+        num, den = num.cancel(den, include=True)
+    num, den = (num*x**mul).cancel(den, include=True)
+
+    c = IndexedBase('c')
+    out, sums = S(0), S(0)
+    for pw, cf in den.as_dict().items():
+        pw = pw[0]
+        for i in range(pw, max(den.degree(x), num.degree(x) + 1)):
+            out += cf*c[i - pw]*x**i
+        sums += cf*c[m - pw]
+
+    if out:
+        coeffs = solve_undetermined_coeffs(Eq(num.expr, out), (out.atoms(Indexed)), x)
+        for i in range(len(coeffs), n + mul):
+            ai = solve_undetermined_coeffs(sums.subs(m, i).subs(coeffs), [c[i]], x)
```

Comment:
What is the advantage of using `solve_undetermined_coeffs` rather than `linsolve?

Reply:
I thought using `solve_undetermined_coeffs` is better since it is more specific, i.e. `linsolve` has other types of equations it can solve

Reply:
You should compare them to see which gives a more usable output and which is faster.

Reply:
Changed to `linsolve` since `solve_undetermined_coeffs` is calling `solve` internally.